packages/device-connector/src/index.ts

```diff
@@ -1,24 +1,22 @@
-import { RPCClient } from './cli-rpc/client'
-import { openStream } from './deployment-server/connection'
-import { deployBinary, setDevices } from './devices/service'
+import { closeConnector, createConnector, DeviceConnector } from './deviceConnector'
+import { exit } from 'node:process'
+import closeWithGrace from 'close-with-grace'
 
-const client = await new RPCClient()
-await client.init()
-await client.createInstance()
-await client.initInstance()
+try {
+  const connector: DeviceConnector = await createConnector()
 
-const socket = await openStream()
+  const handler = closeWithGrace({ delay: 1000 }, closeConnector.bind(connector))
 
-await client.boardListWatch()
-const ourDevices = client.getDevices()
-setDevices(ourDevices)
-
-socket.onmessage = (e) => {
-  const data = typeof e.data === 'string' ? JSON.parse(e.data) : e.data
+  // Nodemon sends SIGUSR2 when it restarts
+  process.once('SIGUSR2', () => {
+    handler.close()
+  })
 
-  deployBinary(data, client).then(() => {
-    console.log('done')
-  }).catch((e) => {
-    console.log(e)
+  // Process SIGINT signal
+  process.once('SIGINT', () => {
```

Comment:
close with grace handles this

Reply:
I encountered a certain case where it does not. Whenever I press the stop process button within my IDE, it terminates the service without closing gracefully. I found that to be a SIGINT signal. Since I am handling this here, it does close as it is supposed to